main/src/library/File.flix

```diff
@@ -119,6 +119,55 @@ namespace File {
                 Err(getMessage(ex))
         }
 
+    ///
+    /// Returns a string containing the given file `f`.
+    ///
+    pub def read(f: String): Result[String, String] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:readString(##java.nio.file.Path);
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let result = readString(javaPath);
+            Ok(result)
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Returns a string containing the given file `f` with the options `opts`.
+    /// The options `opts` to apply consists of
+    /// `offset` - the start offset in the given file `f`.
+    /// `count` - the number of bytes read.
+    /// `charSet` - the specific charset to be used to decode the bytes.
+    ///
+    pub def readWith(opts: {offset :: Int32, count :: Int32, charSet :: String}, f : String): Result[String, String] & Impure = {
+        try {
+            import new java.io.File(String) as newFile;
+            import new java.io.FileInputStream(##java.io.File) as newInputStream;
+            import java.io.FileInputStream.read(Array[Int8], Int32, Int32);
+            import java.nio.charset.Charset:forName(String);
+            import new java.lang.String(Array[Int8], ##java.nio.charset.Charset) as newString;
+
+            let javaFile = newFile(f);
+            let stream = newInputStream(javaFile);
+            let bytes = [0i8; 10];
+            // TODO: I get IndexOutOfBoundsException when reading if: let bytes = [];
+            // TODO: "enter" read with offset = 3, count = 3 -> "ent"
```

Comment:
I don't know how the bytes array for the read should be initialized

Reply:
I needs to be able to contain all the read bytes, so it should have size `count`.

Another thing is that `FileInputStream::read` returns an integer "the total number of bytes read into the buffer, or -1 if there is no more data because the end of the file has been reached." If this number is `!= count` then some of the string will be random bytes. So if this is the case an error should maybe just be returned? It has to be handled but im not sure how.